graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/tracing/FederatedTracingInstrumentation.java

```diff
@@ -197,14 +198,14 @@ public InstrumentationState createState(InstrumentationCreateStateParameters par
     private static class FederatedTracingState implements InstrumentationState {
         private final Instant startRequestTime;
         private final long startRequestNanos;
-        private final LinkedHashMap<ExecutionPath, Reports.Trace.Node.Builder> nodesByPath;
+        private final ConcurrentHashMap<ExecutionPath, Reports.Trace.Node.Builder> nodesByPath;
```

Comment:
I wonder if we should just call this a `Map` here? Do we use anything specific to concurrent maps?

Reply:
We could, although I'd rather explicitly require that the map is intended to be thread-safe via types.

I'm fine with it being `ConcurrentMap`.